### PR TITLE
Mbarba/sift fixes release 93

### PR DESCRIPTION
**Pull request by MatBarba**
General fixes for Sift to run more smoothly, in particular to ignore the "failed" runsift jobs that are actually expected (see commits for detail). Tested on Vectorbase databases.

### CODE REVIEW

File: modules/Bio/EnsEMBL/Variation/Pipeline/ProteinFunction/RunSift.pm [hunk 1]
```diff
@@ -136,6 +137,18 @@ sub run {
...
       else {
+        # If we just did not find enough hits, that's ok, just skip this
+        my $error_file = $fasta_file . ".query.globalX.error";
+        if (-s $error_file) {
+          open my $error_fh, "<", $error_file or die $!;
+          my $error_msg = <$error_fh>;
+          close $error_fh;
+          chomp $error_msg;
+          if ($error_msg =~ /Not enough sequences \(only \d\) found by the PSI-BLAST search!|PSI-BLAST found no hits/) {
+            return;
```
Comment (at7) [inline modules/Bio/EnsEMBL/Variation/Pipeline/ProteinFunction/RunSift.pm]: We still want to get this information , not enough sequence , somewhere. Can that be written as a warning message into the log table? Let's have someone else's opinion too.

File: modules/Bio/EnsEMBL/Variation/Pipeline/ProteinFunction/RunSift.pm [hunk 2]
```diff
@@ -176,7 +189,23 @@ sub run {
...
-    die("Failed to run $flat_cmd: $stderr\n") unless $exit_code == 0;
+    if ($exit_code != 0) {
+      # If there was not enough sequences selected, skip it
+      my $error_file = "protein.alignedfasta.error";
+      if (-s $error_file) {
+        open my $error_fh, "<", $error_file or die $!;
+        my $error_msg = <$error_fh>;
+        close $error_fh;
+        chomp $error_msg;
+        if ($error_msg =~ /\d sequence\(s\) were chosen\. Less than the minimum number of sequences required/) {
```
Comment (at7) [inline modules/Bio/EnsEMBL/Variation/Pipeline/ProteinFunction/RunSift.pm]: Same as above. Do want to keep track of that?

**Comment by ens-lgil** [comment]
It would be better if the PR is sent to the branch `postreleasefix/93`, in order to easily propagate the fix on both `release/93` and `master` branches

**Comment by MatBarba** [comment]
I moved it to the postreleasefix/93 branch

**Pull request merged**